### PR TITLE
refactor: Refactored codeblock and fixed storybook

### PR DESCRIPTION
**Pull request by RamiAwar**

### CODE REVIEW

File: frontend/src/components/Conversation/CodeBlock.tsx [hunk 1]
```diff
@@ -65,40 +62,52 @@ function getNewCursorPosition(
...
     newJson: string,
     newCreatedAt: string
   ) => void;
   forChart: boolean;
 }) => {
-  const { conversationId } = useParams({ from: "/_app/chat/$conversationId" });
-
   const [savedCode, setSavedCode] = useState<string>(() =>
-    formattedCodeOrInitial(code)
+    formattedCodeOrInitial(code, dialect as SupportedFormatters)
```
Comment (anthony2261) [inline frontend/src/components/Conversation/CodeBlock.tsx:107]: What if the dialect is not supported? Like snowflake

**Review by anthony2261 — APPROVED** [review]
Didn't check the storybook stuff though

File: frontend/src/components/Conversation/CodeBlock.tsx (hunk 1 above)
Comment (RamiAwar) [reply frontend/src/components/Conversation/CodeBlock.tsx:107]: Not sure what would happen, gotta try it

**Comment by RamiAwar** [comment]
> Didn't check the storybook stuff though

it's okay, nothing fancy there

**Pull request merged**